crmsh/utils.py

```diff
@@ -2287,4 +2253,58 @@ def interface_choice():
     # should consider interface format like "ethx@xxx"
     interface_list = re.findall(r'(?:[0-9]+:) (.*?)(?=: |@.*?: )', out)
     return [nic for nic in interface_list if nic != "lo"]
+
+
+def interface_info(ipv6=False):
+    """
+    Get network information(IPv4 default) in local, include:
+        nic-ip dictionary
+        ip list
+        ip with mask list
+    IPv6 support
+    """
+    nic_ip_dict = {}
+    ip_list = []
+    ip_with_mask_list = []
+
+    cmd = "ip -{} -o addr show".format(6 if ipv6 else 4)
+    rc, out, err = get_stdout_stderr(cmd)
+    if rc != 0:
+        raise ValueError(err)
+
+    for line in out.split('\n'):
+        _, nic, _, ip_with_mask, *_ = line.split()
+        if nic == "lo":
+            continue
+        ip, _ = ip_with_mask.split('/')
+        ip_list.append(ip)
+        ip_with_mask_list.append(ip_with_mask)
+        # get first address on each nic
+        if nic not in nic_ip_dict:
+            nic_ip_dict[nic] = ip
+    return nic_ip_dict, ip_list, ip_with_mask_list
+
+
+def get_network(ip_with_mask):
+    """
+    Get IP address's network
+    IPv6 support
+    """
+    return str(ipaddress.ip_interface(ip_with_mask).network.network_address)
+
+
+def get_default_nic_from_route():
+    """
+    Get default nic name from default route
+    Return None if route not configured
+    """
+    cmd = "ip -o route show"
+    rc, out, err = get_stdout_stderr(cmd)
+    if rc != 0:
+        raise ValueError(err)
+    res = re.search(r'^default via .* dev (.*?) proto', out)
+    if res:
+        return res.group(1)
+    else:
```

Comment:
In this case, is better to raise an exception. Not having a default is an exception. Better to treat this way instead of returning `None`

Reply:
I prefer to raise an exception and treat that exception as a warning, not an error, since configure could be continued